src/multibase.js

```diff
@@ -53,16 +48,13 @@ function decode (bufOrString) {
     bufOrString = new Buffer(bufOrString)
   }
 
-  let decode
+  const base = getBase(code)
 
-  switch (code) {
-    case 'z': {
-      decode = (buf) => { return new Buffer(bs58.decode(buf.toString())) }
-    } break
-    default: throw errNotSupported
+  const decodeObject = {
+    base: base.getName(),
+    data: new Buffer(base.decode(bufOrString.toString()))
   }
-
-  return decode(bufOrString)
+  return decodeObject.data
```

Comment:
I'm fine with that, although I'm not sure this is complying with the comment above : 
```
// receives a buffer or string encoded with multibase header
// decodes it and returns an object with the decoded buffer
// and the encoded type { base: <name>, data: <buffer> }ments made above the function :
```
I didn't want to change the behavior but I feel the expected output in the test is wrong and this comment is right (look at the GO implementation on this : [here](https://github.com/multiformats/go-multibase/blob/master/multibase.go#L36) (couldn't find any actual spec regarding this)